src/algebra/algebra/subalgebra.lean

```diff
@@ -735,6 +736,80 @@ set_like.coe_injective set.prod_inter_prod
 
 end prod
 
+section supr_lift
+variables {ι : Type*}
+
+lemma coe_supr_of_directed [nonempty ι] {S : ι → subalgebra R A}
+  (dir : directed (≤) S) :
+  ((supr S : subalgebra R A) : set A) = ⋃ i, (S i : set A) :=
```

Comment:
Shorter but maybe not clearer as
```suggestion
  ↑(supr S) = ⋃ i, (S i : set A) :=
```

Reply:
This is part of #9019, so I'll make the changes there and merge that branch here.